src/org/omegat/gui/editor/EditorController.java

```diff
@@ -1456,6 +1456,40 @@ public void nextTranslatedEntry() {
         nextTranslatedEntry(true);
     }
 
+    private void linkedEntry(boolean forward, String linked) {
+        iterateToEntry(forward, ste -> {
+                TMXEntry info = Core.getProject().getTranslationInfo(ste);
+                if (String.valueOf(info.linked) == linked) {
```

Comment:
What are you doing here?

```
                if (String.valueOf(info.linked) == linked) {
```

Reply:
Where is a test code of this? Please put unit test for the additions.
I cannot understand why it works correctly.

Reply:
I have very little idea about tests, but I'll try to look into it. I don't have time till the end of the week, but I'll try to resolve it asap.

Reply:
> @kosivantsov please fix SpotBugs error. SpotBugs is static souce code analyzer and its error means your program code has a bug.
> 
> You can run SpotBugs with `./gradlew spotbugsMain` then you will find a message like
> 
> ```
> Execution failed for task ':spotbugsMain'.
> > A failure occurred while executing com.github.spotbugs.snom.internal.SpotBugsRunnerForWorker$SpotBugsExecutor
>    > 1 SpotBugs violations were found. See the report at: file:///home/kos/omegat/build/reports/spotbugs/main.html
> ```
> 
> Then read report on web browser.

FYI. A common idiom for conditionals of null-able string in OmegaT is `if (!StringUtil.isEmpty(a) && a.equals(b)) {...}`. 

I don't know why you need to use a special equal operator "==" , and it becomes a fatal error on SpotBugs static code analysis and quality assurance tool.